src/components/Editor/index.js

```diff
@@ -418,7 +418,13 @@ class Editor extends PureComponent<Props, State> {
     const { editor } = this.state;
 
     if (this.shouldScrollToLocation(nextProps)) {
-      const { line, column } = toEditorPosition(nextProps.selectedLocation);
+      let { line, column } = toEditorPosition(nextProps.selectedLocation);
+
+      const lineText = editor.editor.getLine(line);
+      const spaces = lineText.match(/^\s+/);
+      if (spaces) {
+        column = Math.max(column, spaces[0].length);
+      }
```

Comment:
lets see if we can reuse the indentation helper:

this way we can write something like `column = Math.max(column, getIndentation(lineText));`

```diff
diff --git a/src/utils/indentation.js b/src/utils/indentation.js
index 8452f5e..579c6b6 100644
--- a/src/utils/indentation.js
+++ b/src/utils/indentation.js
@@ -2,7 +2,17 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
-function getIndentation(lines) {
+// @flow
+
+export function getIndentation(line?: string) {
+  if (!line) {
+    return 0;
+  }
+
+  return line.match(/^\s*/)[0].length;
+}
+
+function getFunctionIndentation(lines) {
   const firstLine = lines[0];
   const secondLine = lines[1];
   const lastLine = lines[lines.length - 1];
@@ -20,7 +30,7 @@ function getIndentation(lines) {
 
 export function correctIndentation(text) {
   const lines = text.trim().split("\n");
-  const indentation = getIndentation(lines);
+  const indentation = getFunctionIndentation(lines);
   const formattedLines = lines.map(_line =>
     _line.replace(new RegExp(`^\\s{0,${indentation - 1}}`), "")
   );
```

Reply:
Okay sounds good, I'll take a look now :)